### PR TITLE
Fix: make run_given_checks work correctly

### PR DESCRIPTION
**Pull request by phlax**
with TeeChecker

and correctly separate translated units in mass checker

### CODE REVIEW

File: pootle/apps/pootle_misc/checks.py [hunk 1]
```diff
@@ -971,7 +971,19 @@ def run_given_filters(checker, unit, check_names=None):
...
+            for _checker in checker.checkers:
+                filterfunction = getattr(_checker, functionname, None)
+                if filterfunction:
+                    checker = _checker
+                    checker.str1 = data.normalized_unicode(unit.source) or u""
+                    checker.str2 = data.normalized_unicode(unit.target) or u""
+                    checker.language_code = unit.language_code
+                    checker.hasplural = unit.hasplural()
+                    checker.locations = unit.getlocations()
+                    break
```
Comment (unho) [inline pootle/apps/pootle_misc/checks.py]: This means that only the first checker in teechecker is run, but the purpose of teechecker is to give the ability to run checks from several checkers. This `break` defuses that.

Comment (phlax) [inline pootle/apps/pootle_misc/checks.py]: it means the first one with a matching check

Comment (unho) [inline pootle/apps/pootle_misc/checks.py]: I see now that the code is correct, despite it only taking the first filter matching the provided name.

Comment (unho) [inline pootle/apps/pootle_misc/checks.py]: Is `filterfunction` unset if none of the checkers has any matching filter?

Comment (phlax) [inline pootle/apps/pootle_misc/checks.py]: yep - and as a consequence any exsting checks would be deleted

Comment (phlax) [inline pootle/apps/pootle_misc/checks.py]: ie

```
pootle calculate_checks --check-name=my_old_check_thats_no_longer_used
```
would delete the corresponding check

**Comment by unho** [comment]
lgtm

**Pull request merged**